### PR TITLE
changes that enable cice6 to run as an external cesm component

### PR DESCRIPTION
**Pull request by mvertens**
For detailed information about submitting Pull Requests (PRs) to the CICE-Consortium,
please refer to: <https://github.com/CICE-Consortium/About-Us/wiki/Resource-Index#information-for-developers>

## PR checklist
- [x] Short (1 sentence) summary of your PR: 
    changes that enable cice6 to run as an external cesm componentENTER INFORMATION HERE
- [x] Developer(s): 
    mvertens
- [x] Suggest PR reviewers from list in the column to the right.
- [x] Please copy the PR test results link or provide a summary of testing completed below.
    aux_cice6 
   This PR needs the accompanying CESM_CICE PR in order to run out of the box CESM tests
- How much do the PR code changes differ from the unmodified code? 
    - [x] bit for bit (this needs to be verified)
    - [ ] different at roundoff level
    - [ ] more substantial 
- Does this PR create or have dependencies on Icepack or any other models?
    - [ ] Yes - 
    - [x] No
- Does this PR add any new test cases?
    - [x] Yes - the aux_cice6 test list
    - [ ] No
- Is the documentation being updated? ("Documentation" includes information on the wiki or in the .rst files from doc/source/, which are used to create the online technical docs at https://readthedocs.org/projects/cice-consortium-cice/. A test build of the technical docs will be performed as part of the PR testing.)
    - [ ] Yes
    - [x] No, does the documentation need to be updated at a later time?
        - [x] Yes
        - [ ] No 
- [ ] Please provide any additional information or relevant details below:

### CODE REVIEW

File: cicecore/drivers/nuopc/cmeps/ice_comp_nuopc.F90 [hunk 1]
```diff
@@ -26,6 +26,7 @@ module ice_comp_nuopc
   use ice_calendar       , only : idate, mday, time, month, daycal, time2sec, year_init
   use ice_calendar       , only : sec, dt, calendar, calendar_type, nextsw_cday, istep
   use ice_kinds_mod      , only : dbl_kind, int_kind, char_len, char_len_long
+  use ice_scam           , only : scmlat, scmlon, scol_mask, scol_frac, scol_ni, scol_nj, scol_valid, single_column
```
Comment (dabail10) [inline cicecore/drivers/nuopc/cmeps/ice_comp_nuopc.F90]: Does this need to be wrapped in a CESMCOUPLED ifdef for EMC?

File: cicecore/drivers/nuopc/cmeps/ice_comp_nuopc.F90 [hunk 2]
```diff
@@ -470,38 +493,111 @@ subroutine InitializeAdvertise(gcomp, importState, exportState, clock, rc)
...
-    !----------------------------------------------------------------------------
-
-    ! Determine the model distgrid using the decomposition obtained in
-    ! call to init_grid1 called from cice_init1
-    call ice_mesh_set_distgrid(localpet, npes, ice_distgrid, rc)
+    ! Form of ocean freezing temperature
+    ! 'minus1p8' = -1.8 C
+    ! 'linear_salt' = -depressT * sss
+    ! 'mushy' conforms with ktherm=2
+    call NUOPC_CompAttributeGet(gcomp, name="tfreeze_option", value=tfrz_option_driver, &
```
Comment (dabail10) [inline cicecore/drivers/nuopc/cmeps/ice_comp_nuopc.F90:505]: Similarly here. Does EMC get tfreeze_option and atmiter_conv from the driver?

**Review by dabail10 — COMMENTED** [review]
I am mostly worried about some of the driver parameters and the single column stuff impacting EMC.

File: cicecore/drivers/nuopc/cmeps/ice_comp_nuopc.F90 (hunk 2 above)
Comment (mvertens) [reply cicecore/drivers/nuopc/cmeps/ice_comp_nuopc.F90:505]: Good point. I think for now these both should be wrapped in a CESMCOUPLED #ifdef.

File: cicecore/drivers/nuopc/cmeps/ice_comp_nuopc.F90 (hunk 1 above)
Comment (mvertens) [reply cicecore/drivers/nuopc/cmeps/ice_comp_nuopc.F90]: done.

File: cicecore/drivers/nuopc/cmeps/ice_comp_nuopc.F90 [hunk 3]
```diff
@@ -39,11 +39,13 @@ module ice_comp_nuopc
 #ifdef CESMCOUPLED
   use shr_const_mod
   use shr_orb_mod        , only : shr_orb_decl, shr_orb_params, SHR_ORB_UNDEF_REAL, SHR_ORB_UNDEF_INT
+  use ice_scam           , only : scmlat, scmlon, scol_mask, scol_frac, scol_ni, scol_nj, scol_valid, single_column
 #endif
   use ice_timers
   use CICE_InitMod       , only : cice_init1, cice_init2
   use CICE_RunMod        , only : cice_run
   use ice_mesh_mod       , only : ice_mesh_set_distgrid, ice_mesh_setmask_from_maskfile, ice_mesh_check
+  use ice_mesh_mod       , only : ice_mesh_init_tlon_tlat_area_hm, ice_mesh_create_scolumn
```
Comment (dabail10) [inline cicecore/drivers/nuopc/cmeps/ice_comp_nuopc.F90:48]: Does the ice_mesh_create_scolumn have to be in an ifdef?

**Comment by dabail10** [comment]
I tested this and it was bfb with my previous version. I am going to go ahead and merge. @DeniseWorthen should check to make sure this works with the EMC code.

**Pull request merged**